soweego/linker/evaluate.py

```diff
@@ -7,41 +7,55 @@
 from pandas import concat
-from sklearn.model_selection import KFold, train_test_split
+from sklearn.model_selection import StratifiedKFold, train_test_split
 
 from soweego.commons import constants, target_database
 from soweego.linker import train, workflow
 
 LOGGER = logging.getLogger(__name__)
-rl.set_option(*constants.CLASSIFICATION_RETURN_SERIES)
 
 
 @click.command()
 @click.argument('classifier', type=click.Choice(constants.CLASSIFIERS))
 @click.argument('target', type=click.Choice(target_database.available_targets()))
 @click.argument('target_type', type=click.Choice(target_database.available_types()))
-@click.option('-b', '--binarize', default=0.1, help="Default: 0.1")
-@click.option('-d', '--dir-io', type=click.Path(file_okay=False), default=constants.SHARED_FOLDER,
-              help="Input/output directory, default: '%s'." % constants.SHARED_FOLDER)
-def cli(classifier, target, target_type, binarize, dir_io):
+@click.option('--single', is_flag=True, help='Compute a single evaluation over all k folds, instead of k evaluations.')
+@click.option('-b', '--binarize', default=0.1, help="Default: 0.1.")
+@click.option('-d', '--dir-io', type=click.Path(file_okay=False), default='/app/shared', help="Input/output directory, default: '/app/shared'.")
+def cli(classifier, target, target_type, binarize, single, dir_io):
     """Evaluate the performance of a probabilistic linker."""
-    precision, recall, fscore, confusion_matrix = k_fold_single_evaluation(
-        constants.CLASSIFIERS[classifier], target, target_type, binarize, dir_io)
-    with open(os.path.join(dir_io, constants.LINKER_PERFORMANCE % (target, target_type, classifier)), 'w') as fileout:
-        fileout.write(
-            f'Precision: {precision}\nRecall: {recall}\nF-score: {fscore}\nConfusion matrix:\n{confusion_matrix}\n')
+    if not single:
+        predictions, p_mean, p_std, r_mean, r_std, fscore_mean, fscore_std = average_k_fold(
+            constants.CLASSIFIERS[classifier], target, target_type, binarize, dir_io)
+        LOGGER.info('Precision: mean = %s; std = %s; Recall: mean = %s; std = %s; F-score: mean = %s; std = %s',
+                    p_mean, p_std, r_mean, r_std, fscore_mean, fscore_std)
+
+        predictions.to_series().to_csv(os.path.join(dir_io, constants.LINKER_EVALUATION_PREDICTIONS %
+                                                    (target, target_type, classifier)), columns=[], header=True)
+        with open(os.path.join(dir_io, constants.LINKER_PERFORMANCE % (target, target_type, classifier)), 'w') as fileout:
+            fileout.write(
+                f'Precision:\n\tmean = {p_mean}\n\tstandard deviation = {p_std}\nRecall:\n\tmean = {r_mean}\n\tstandard deviation = {r_std}\nF-score:\n\tmean = {fscore_mean}\n\tstandard deviation = {fscore_std}\n')
+    else:
+        predictions, (precision, recall, fscore, confusion_matrix) = single_k_fold(
+            constants.CLASSIFIERS[classifier], target, target_type, binarize, dir_io)
```

Comment:
if we do decide to let users specify the `k` parameter then it should also be added here